matrix/Matrix.hpp

```diff
@@ -290,8 +290,8 @@ class Matrix
 
     void print() const
     {
-        char buf[200];
-        write_string(buf, 200);
+        char buf[1000];
```

Comment:
That's pretty evil on an embedded system. We need to find a better way than this.